pkg/ccl/backupccl/restore_job.go

```diff
@@ -950,6 +1010,11 @@ func createImportingDescriptors(
 					}
 					typDesc := desc.(*typedesc.Mutable)
 					typDesc.AddReferencingDescriptorID(table.GetID())
+					// TODO (lucy): I think we should be incrementing the version for the
```

Comment:
maybe this stuff too

Reply:
Is it valid to add the table descriptors (even if offline) but not add the types they reference at the same time?

Reply:
hmm, probably not. We need to have those backreferences there to avoid dropping the type as soon as we add the table descriptor to the database.